doc/api/n-api.md

```diff
@@ -4785,6 +4785,88 @@ with the resulting JavaScript constructor (which is returned in the `result`
 parameter) and freed whenever the class is garbage-collected by passing both
 the JavaScript function and the data to [`napi_add_finalizer`][].
 
+### napi_define_subclass
+<!-- YAML
+added: REPLACEME
+-->
+
+```c
+napi_status napi_define_subclass(napi_env env,
+                                 napi_value parent_constructor,
+                                 const char* utf8name,
+                                 size_t length,
+                                 napi_callback constructor,
+                                 napi_callback get_super_params,
+                                 void* data,
+                                 size_t property_count,
+                                 const napi_property_descriptor* properties,
+                                 napi_value* result);
+```
+
+* `[in] env`: The environment that the API is invoked under.
+* `[in] parent_constructor`: Constructor of the parent class.
+* `[in] utf8name`: Name of the JavaScript constructor function; this is
+  not required to be the same as the C++ class name, though it is recommended
```

Comment:
This is talking about C++ everywhere, but C is what we mean, right?

Reply:
Yeah, I copied the doc mostly from `napi_define_class`, updating for the different functionality. I suppose it might be worth updating the doc for `napi_define_class` to not assume folks write C++ classes. I'd like to place that beyond the scope of this PR though.

https://github.com/nodejs/node/issues/36150